src/main/java/org/craftercms/studio/impl/v2/upgrade/operations/BlueprintRenameFolderUpgradeOperation.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2007-2019 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.craftercms.studio.impl.v2.upgrade.operations;
+
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.io.FileUtils;
+import org.craftercms.studio.api.v1.constant.GitRepositories;
+import org.craftercms.studio.api.v1.log.Logger;
+import org.craftercms.studio.api.v1.log.LoggerFactory;
+import org.craftercms.studio.api.v1.service.configuration.ServicesConfig;
+import org.craftercms.studio.api.v1.service.security.SecurityService;
+import org.craftercms.studio.api.v2.exception.UpgradeException;
+import org.craftercms.studio.api.v2.service.security.internal.UserServiceInternal;
+import org.craftercms.studio.impl.v1.repository.git.GitContentRepositoryHelper;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.Status;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.lib.Repository;
+import org.springframework.beans.factory.annotation.Required;
+import org.springframework.web.context.ServletContextAware;
+
+import javax.servlet.ServletContext;
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import static org.craftercms.studio.api.v1.util.StudioConfiguration.BLUE_PRINTS_PATH;
+import static org.craftercms.studio.api.v1.util.StudioConfiguration.REPO_BLUEPRINTS_UPDATED_COMMIT_MESSAGE;
+import static org.craftercms.studio.impl.v1.repository.git.GitContentRepositoryConstants.GIT_COMMIT_ALL_ITEMS;
+
+/**
+ * Implementation of {@link org.craftercms.studio.api.v2.upgrade.UpgradeOperation} that renames the blueprint in the
+ * global repository.
+ * @author Dejan Brkic
+ */
+public class BlueprintRenameFolderUpgradeOperation extends AbstractUpgradeOperation implements ServletContextAware {
```

Comment:
- `AbstractUpgradeOperation` already implements `ServletContextAware`
- Wouldn't it be better if we make a generic `RenameFolderUpgradeOperation` in case we need it later?

Reply:
I took `BlueprintsUpgradeOperation` as the starting point. Has the same extends and implements. I will fix them both.
At the moment I don't see a use case that needs the generic operation.

Reply:
Yes `BlueprintsUpgradeOperation` was left behind after some refactors, if you can update it also it would be great.

But I still think it would be better to make it as generic as possible so we don't end with a lot of specific classes and even if we don't needed it now we might need to rename folders or files inside the site repositories in the future, @sumerjabri what do you think about this?

Reply:
Let's do a generic `RenameOperation` so we can rename anything. Any issues with that?